src/stdio.jl

```diff
@@ -52,6 +78,34 @@ function watch_stream(rd::IO, name::AbstractString)
     end
 end
 
+function send_stdio(name)
+    if verbose::Bool && !haskey(task_local_storage(), :IJulia_task)
+        task_local_storage(:IJulia_task, "send $name task")
+        @vprintln("$name task labelled")
+    end
+    send_stream(name)
+end
+
+send_stdout(t::Timer) = send_stdio("stdout")
+send_stderr(t::Timer) = send_stdio("stderr")
+
+function send_stream(name::AbstractString)
+    buf = bufs[name]
+    if buf.size > 0
+        s = takebuf_string(buf)
+        if isvalid(s)
+            send_ipython(publish,
+                     msg_pub(execute_msg, "stream",
+                             @compat Dict("name" => name, "text" => s)))
+        else
+            #if the string is not valid, maybe it's the start of a partially encoded UTF character
+            #so we write it back to the buffer and hope the rest of the character will arrive soon
+            @vprintln("Invalid UTF string written to ", name, ". Bytes: ", convert(Vector{UInt8}, s))
```

Comment:
I don't like having debugging printfs permanently in the code...

Also, this solution is neither efficient nor robust.  It is not efficient because it would be better just to check the end of the string for a partially encoded UTF-8 character, and write only *that* part of the data back to the buffer, sending the rest of the string to Python.   It's not robust because it doesn't handle the case where the use just calls `write(STDOUT, ....raw binary data that is not UTF-8....)`, in which case we should still probably send *something* to IPython.